lib/polymorphic_integer_type/belongs_to_polymorphic_association_extension.rb

```diff
@@ -3,7 +3,7 @@ module Associations
     class BelongsToPolymorphicAssociation < BelongsToAssociation
       private def replace_keys(record)
         super
-        owner[reflection.foreign_type] = record.class
+        owner[reflection.foreign_type] = record&.class&.base_class
```

Comment:
record can be nil which causes lots of failures in 5.2. With safe operator our specs pass.

Reply:
Does it matter that we create a new key,value pair here for `(type, nil)`, or should we do nothing if `record` is `nil`?